### PR TITLE
Cancel GetDiscoveredNodesAction when bootstrapped

### PR DESCRIPTION
**Pull request by DaveCTurner**
Today the `GetDiscoveredNodesAction` waits, possibly indefinitely, to discover
enough nodes to bootstrap the cluster. However it is possible that the cluster
forms before a node has discovered the expected collection of nodes, in which
case the action will wait indefinitely despite the fact that it is no longer
required.

This commit changes the behaviour so that the action fails once a node receives
a cluster state with a nonempty configuration, indicating that the cluster has
been successfully bootstrapped and therefore the `GetDiscoveredNodesAction`
need wait no longer.

Relates #36380 and #36381; reverts 558f4ec27820e1a50660dc1f3437422150339af0.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/action/admin/cluster/bootstrap/TransportGetDiscoveredNodesAction.java [hunk 1]
```diff
@@ -113,15 +123,18 @@ public String toString() {
...
+            respondIfRequestSatisfied.onFailure(new ClusterAlreadyBootstrappedException());
+        } else {
+            respondIfRequestSatisfied.onResponse(coordinator.getFoundPeers());
+        }
 
         if (request.getTimeout() != null) {
             threadPool.schedule(request.getTimeout(), Names.SAME, new Runnable() {
                 @Override
                 public void run() {
-                    if (listenerNotified.compareAndSet(false, true)) {
```
Comment (ywelsch) [inline server/src/main/java/org/elasticsearch/action/admin/cluster/bootstrap/TransportGetDiscoveredNodesAction.java]: why is this not needed anymore?

File: server/src/main/java/org/elasticsearch/action/admin/cluster/bootstrap/GetDiscoveredNodesAction.java [hunk 2]
```diff
@@ -23,7 +23,7 @@
 
 public class GetDiscoveredNodesAction extends Action<GetDiscoveredNodesResponse> {
     public static final GetDiscoveredNodesAction INSTANCE = new GetDiscoveredNodesAction();
-    public static final String NAME = "cluster:monitor/discovered_nodes";
+    public static final String NAME = "cluster:admin/get_bootstrap_discovered_nodes";
```
Comment (ywelsch) [inline server/src/main/java/org/elasticsearch/action/admin/cluster/bootstrap/GetDiscoveredNodesAction.java]: maybe `cluster:admin/bootstrap/discover_nodes` and call the other one `cluster:admin/bootstrap/set_voting_config` and then whitelist `cluster:admin/bootstrap/*`?

**Review by ywelsch — APPROVED** [review]

File: server/src/main/java/org/elasticsearch/action/admin/cluster/bootstrap/TransportGetDiscoveredNodesAction.java (hunk 1 above)
Comment (DaveCTurner) [reply server/src/main/java/org/elasticsearch/action/admin/cluster/bootstrap/TransportGetDiscoveredNodesAction.java]: `respondIfRequestSatisfied.onFailure()` checks this.

File: server/src/main/java/org/elasticsearch/action/admin/cluster/bootstrap/GetDiscoveredNodesAction.java (hunk 2 above)
Comment (DaveCTurner) [reply server/src/main/java/org/elasticsearch/action/admin/cluster/bootstrap/GetDiscoveredNodesAction.java]: Thanks. 981dcc0434c8db3eff72ab9c50999710f13d4520.

File: server/src/main/java/org/elasticsearch/action/admin/cluster/bootstrap/TransportGetDiscoveredNodesAction.java (hunk 1 above)
Comment (ywelsch) [reply server/src/main/java/org/elasticsearch/action/admin/cluster/bootstrap/TransportGetDiscoveredNodesAction.java]: ok thanks

**Pull request merged**